[code-infra] Remove modern bundles (#45808)
mui/material-ui: docs/data/material/migration/upgrade-to-v7/upgrade-to-v7.md

This was never officially supported, but now it will be restricted by bundlers and runtimes.

> The modern bundles have also been removed. It turned out that the potential win in bundle size is insignificant nowadays. If you've configered aliases for these bundles, you will now have to remove them.

Reviewer: ```suggestion
Modern bundles have also been removed, as the potential for a smaller bundle size is no longer significant.
If you've configured aliases for these bundles, you must remove them now.
```

```diff
 {
   resolve: {
     alias: {
-      '@mui/material': '@mui/material/modern',
-      '@mui/styled-engine': '@mui/styled-engine/modern',
-      '@mui/system': '@mui/system/modern',
-      '@mui/base': '@mui/base/modern',
-      '@mui/utils': '@mui/utils/modern',
-      '@mui/lab': '@mui/lab/modern',
     }
   }
 }
```

:::info

> Earlier versions of this guide mention the existence of a `mui-modern` conditional exports. This has since been removed. This is a non-breaking change, your bundler will fall back to the ESM bundle.

Reviewer: ```suggestion
Earlier versions of this guide mention the existence of a `mui-modern` conditional exports. 
This has since been removed. 
This is a non-breaking change, and your bundler will fall back to the ESM bundle.
```
